src/components/Swap/SwapButton/index.tsx

```diff
@@ -67,80 +67,73 @@ export default memo(function SwapButton({ disabled }: SwapButtonProps) {
   // Close the review modal on chain change.
   useEffect(() => setOpen(false), [chainId])
 
-  const addTransaction = useAddTransaction()
+  const addTransactionInfo = useAddTransaction()
   const setDisplayTxHash = useUpdateAtom(displayTxHashAtom)
-  const setOldestValidBlock = useSetOldestValidBlock()
+  const [, setSwapAmount] = useSwapAmount(Field.INPUT)
+  const resetSwapAmount = useCallback(() => setSwapAmount(''), [setSwapAmount])
+
+  const onSubmit = useCallback(
+    async (submit: () => Promise<TransactionInfo | undefined>): Promise<void> => {
+      try {
+        const info = await submit()
+        if (!info) return
+
+        resetSwapAmount()
+        addTransactionInfo(info)
+        setDisplayTxHash(info.response.hash)
+
+        if (isAnimating(document)) {
+          // Only return after any queued animations to avoid layout thrashing, because a successful
+          // submit will open the status dialog and immediately cover the animating button.
+          return new Promise((resolve) => {
+            const onAnimationEnd = () => {
+              document.removeEventListener('animationend', onAnimationEnd)
+              resolve()
+            }
+            document.addEventListener('animationend', onAnimationEnd)
+          })
+        }
+      } catch (e) {
+        console.error('Failed to submit', e)
+      }
+    },
+    [addTransactionInfo, resetSwapAmount, setDisplayTxHash]
+  )
 
   const [isPending, setIsPending] = useState(false)
   const native = useNativeCurrency()
   const onWrap = useCallback(async () => {
     setIsPending(true)
-    try {
-      const transaction = await wrapCallback()
-      if (!transaction) return
-      invariant(wrapType !== undefined)
-      addTransaction({
-        response: transaction,
-        type: wrapType,
-        amount: CurrencyAmount.fromRawAmount(native, transaction.value?.toString() ?? '0'),
-      })
-      setDisplayTxHash(transaction.hash)
-      // Only reset pending after any queued animations to avoid layout thrashing, because a
-      // successful wrap will open the status dialog and immediately cover the button.
-      const postWrap = () => {
-        setIsPending(false)
-        document.removeEventListener('animationend', postWrap)
-      }
-      if (isAnimating(document)) {
-        document.addEventListener('animationend', postWrap)
-      } else {
-        setIsPending(false)
-      }
-    } catch (e) {
-      // TODO(zzmp): Surface errors from wrap.
-      console.log(e)
-    } finally {
-      setIsPending(false)
```

Comment:
is this not necessary anymore?

Reply:
`onSubmit` will always return, so a `finally` is no longer needed.